### PR TITLE
Use the fdb cluster version for the backup version

### PR DESCRIPTION
**Pull request by johscheuer**
Fixes: https://github.com/FoundationDB/fdb-kubernetes-operator/issues/168

### CODE REVIEW

**Comment by brownleej** [comment]
How will this behave when we upgrade the cluster? Will it automatically trigger a reconciliation of the backup?

**Comment by johscheuer** [comment]
> How will this behave when we upgrade the cluster? Will it automatically trigger a reconciliation of the backup?

I have to validate this (I'm not sure if the Backup controller watches the clusters) but it will reconcile either with a watch event or the time trigger.

**Comment by brownleej** [comment]
I've also been wondering if the idea of using the cluster spec to determine the configuration for the backup deployment is a little abstraction-breaking, since we now have this deployment based on a composite of two objects.

**Comment by johscheuer** [comment]
I think the main question here is do we want to support Backups of non Kubernetes clusters (which probably don't have a cluster spec) otherwise I think this approach is reasonable and we currently use the cluster spec anyway.

File: controllers/update_backup_agents.go [hunk 1]
```diff
@@ -54,7 +54,12 @@ func (u UpdateBackupAgents) Reconcile(r *FoundationDBBackupReconciler, context c
 		}
 	}
 
-	deployment, err := GetBackupDeployment(backup)
+	version, err := r.getBackupVersion(context, backup)
```
Comment (johscheuer) [inline controllers/update_backup_agents.go:57]: TODO: see if we can reconcile if the cluster status changes.

**Comment by johscheuer** [comment]
Closing since we decided not to add this dependency

**Pull request closed without merging**